MS-3 Úložiště zdrojových 3D modelů

AdamPospisil opened the pull request:
https://stdio.youtrack.cloud/issue/MS-3/Uloziste-zdrojovych-3D-modelu

AdamPospisil committed:
refactor: split models/index into separate queries and mutations

AdamPospisil committed:
fix: models imports in api

AdamPospisil committed:
feat: models frontend WIP

AdamPospisil committed:
feat: frontend model upload & delete

AdamPospisil committed:
redirect on model upload

AdamPospisil committed:
Merge branch 'dev' into feature/storage_frontend

AdamPospisil committed:
feat: model file download

AdamPospisil committed:
feat: refetch models on model deletion

AdamPospisil committed:
fix: imports in model tests

AdamPospisil committed:
feat: model pages require enabled user WIP

vojtatom committed:
fix: add correct typing to HOC withUserEnabled

vojtatom reviewed (changes requested):
Vypadá to super!

- přidat klik na drag and drop areu, která otevře file select dialog
- upravit layout formuláře - nejprve drag and drop, pak seznam souborů
- na upload přidat indikátor že se soubor odesílá/je na to tam ta `isLoading` flag
- kouknout na ošetření situace kdy je prázdné jméno souboru prázdné, nebo nejsou vybrány žádné soubory

AdamPospisil committed:
feat: model upload form validation

AdamPospisil committed:
refactor: added some comments to model upload page

AdamPospisil committed:
feat: made mode upload into a dialog instead its own page

AdamPospisil committed:
model detail dialog

AdamPospisil committed:
Merge branch 'dev' into feature/storage_frontend

AdamPospisil committed:
refactor: added toPlain to models mutations and queries

AdamPospisil committed:
feat: model upload form validation uses toast to display messages

AdamPospisil committed:
feat: moved model detail page to modal dialog

AdamPospisil committed:
refactor: delete model is now client hook instead of api endpoint

AdamPospisil committed:
feat: model delete has to be confirmed in dialog window

SmallhillCZ reviewed (changes requested):
.

SmallhillCZ reviewed (commented):
Nedaří se mi stáhnout nahraný model (není to v user story, ale je tam na to tlačítko)

AdamPospisil committed:
fix: open model detail dialog only once the model instance is fetched

AdamPospisil committed:
feat: download all model files as an archive

AdamPospisil committed:
Merge branch 'dev' into feature/storage_frontend

AdamPospisil committed:
Merge branch 'dev' into feature/storage_frontend

vojtatom committed:
chore: remove Own from naming

vojtatom committed:
feat: rename useOwnModels to useModels

The `useOwnModels` hook has been renamed to `useModels` to improve clarity and consistency in naming conventions. This change ensures that the hook name accurately reflects its purpose.

vojtatom committed:
Merge remote-tracking branch 'origin/dev' into feature/storage_frontend

vojtatom committed:
fix: renamed Own methods
